Return showing duration days back

timur987 opened the pull request:
Related to `Tumblr/a8c-dsp/issues/2896`

## Proposed Changes
Previously we had the number of campaigns days and date ranges shown. Then, in 3fcd217255d07b960e163fccd5f0ce893aab7cd4 we hid this data. Now it's time to start showing it once again.

In particular, we need to show the following (as on the image below):

- duration in days: `Duration: 5 days`
- duration from-to: `Run between: Dec 5 - Dec 10`
- tooltip with time from-to: `Dec 5, 19:50 – Dec 10, 19:50`

![395144913-7c682621-1891-4365-b64b-90f483e0c8b9](https://github.com/user-attachments/assets/39202da2-ca0e-4819-ad50-8c9b65f6ae20)

Note, in addition to "Duration" and "Run between" sections, I've renamed "Clicks" to "Visitors" and "Impressions" to "People reached" as well to be in comply with the latest design from Bart.

## Testing Instructions
- For testing you need to have DSP Widget up and running locally, sandboxed WPCOM and everything we usually do in order to run Calypso with DSP Widget locally,
- In addition to above, we need to have the proper data for campaigns, so please follow instructions from `Tumblr/a8c-dsp/pull/2793`:
  - Create a campaign and change it's ID to become `110111`,
  - Import data from the mentioned PR,
  - Make sure that calling `http://localhost:3003/api/v1.1/stats/110111?start_date=2024-08-23&end_date=2024-08-27&tz=Europe/London` returns data for you,
  - Adjust start_date and end_date in DSP DB for your campaign,
- Open details for campaign `110111`, it should show data properly:

<img width="1091" alt="image" src="https://github.com/user-attachments/assets/f64ff940-f6c9-4f75-b2fb-47dc9e487e15" />

- Popover should work as well:

<img width="600" alt="image" src="https://github.com/user-attachments/assets/cfab8b82-3f22-4713-993c-126ca0dae7ed" />

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

aidvu committed:
Update Playwright to 1.48.2 (#97275)

We've been running an old Playwright, which uses Chromium 116 (Aug 2023). Not a great thing for e2e tests.

Trying to bump to a recent one (1.49.1 fails, so I'm guessing a regression). 1.48.2 is Chromium 130 (Oct 2024), so should work and all pass.

We need to manually install browsers now tho: https://github.com/microsoft/playwright/releases/tag/v1.38.0

aidvu committed:
e2e tests: Firefox for Woo tests (#97335)

aidvu committed:
E2E Tests: Fix LOHP tests (#97338)

Scrolling list of themes needs to be hovered on before click, otherwise
not considered stable by Playwright.

chriskmnds committed:
Plans Step: TS and function form (#96786)

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

aidvu committed:
E2E Tests: Lower workers from 100% to 50% (#97334)

This means out of 16 cores, jest will use 8. Might be better, might be worse, but worth a try if CPU is capped at 100%.

Aurorum committed:
Purchases: Make Refund Window Clearer  (#89249)

* Update context-links.js

* Include refund information

* Update strings

* Address review

* Fix string

* Address review

* Address review

* Address review

* Remove support link

* Simply switch to "Refund available"

* Fix style

* Update style.scss

robfelty committed:
Use the default version of Odie specified in the backend (#97333)

robfelty committed:
Help Center Experience: ramp to 75% of users (#97271)

epeicher committed:
Performance Profiler: Pass locale to LLM Query (#97285)

* Performance Profiler: Pass locale to LLM Query

* Add locale slug to the LLM react query key

* Use the builtin _locale query param when using wpcom requests

It is not necessary to use an explicit locale parameter

valterlorran committed:
Launches the Application Password flow (#97345)

sixhours committed:
Site Migration: Update Plans page for migrations flow (#97062)

* Initial start at new plans page for migrations

* Update loader styles/markup

* Fix header styles

* Remove descriptions

* Fix spacing

* Refactor pricing

* Fix typing

* Fix checkout ctas, update wording for 2 year plan

* Simplify loader

* Simplify pricing output

* Tweak copy, remove unused function param

* Simplify rendering

* Add vars for repeated constants

* Revert changes to Skeleton

* Revert unnecessary name change

* Fix tabs

* Fix for loader display

* Simplify

* Prevent flash of uncentered loader

* Update language for consistency

* More specific typing, pass planSlugs through so we don't have to repeat

* Use feature flag being introduced in separate PR

* Use consistent variable

* Ensure we have planSlugs before reducer

* Update test

* Fix mobile spacing, update pricing copy

* Mobile refund text

* Fix flash of original skeleton

* Fix box shadow on secondary buttons

* Don't show storage

* Use styles to make first list item bold, tweaks to spacing

* No bold text for feature flag

* Add refund tooltip

* Account for border width

* Update spacing under header

* Fix title spacing

* Bold around days

* New strings for features

* Separate Jetpack and WPcom features

* Fix jetpack logo color

* Logo spacing

* Move so we can take full advantage of hooks in translate, add bold text

* Revert changes to plan feature list

* Remove unused prop

* Fix button colors

* Alignment issues

* Fix alignment issues

* Update loader colors to gray

* Calculate percentage rather than hard-coding, consolidate features

a8ck3n committed:
Stats: Gate currently open summary pages (#97274)

* Add missing modules to gated stats

* Add constant for file downloads

* Add new upsell copy

* Reorder switch cases

* Update upsell copy

* Maintain current behaviour for gated modules

aneeshd16 committed:
Add events for opening interval dropdowns (#97309)

* Add events for opening interval dropdowns

* Remove useMemo for hasWPCOMPlanInCart

obenland committed:
Notifications: Restore viewbox for Reblog icon (#97357)

Fixes the display of Reblog gridicons in Notifications.

Introduced in #79932.

dsas committed:
Holiday snow: Pass siteIsWpcom prop (#97363)

This prop was not being passed when the setting is used on tangled calypso  and so the holiday snow setting panel was not being displayed.

dlind1 committed:
Add support for developer.wordpress.com in localizeUrl (#96655)

sirbrillig committed:
Checkout: Make PayPal PPCP available to everyone if server allows it (#97350)

* Remove checkout/paypal-ppcp config option

* Remove "a8c-only" from PayPal PPCP button

* Do not show both paypal-express and paypal-js in checkout

* Only mark PPCP enabled if enabled, not just existing

andrii-lysenko committed:
A4A: fix overdue invoice banner typo (#97303)

yashwin committed:
A4A > Feedback: Handle save feedback (#97106)

* A4A: automated feedbacks UI

* show feedback after some actions are performed

* logic changes

* show notification only if feedback is not shown

* logic update

* initial API integration

* wire up to existing endpoint

* show success notice when feedback is successfully sent

* lint fix

* update success notice copy based on demo video in p2

---------

Co-authored-by: Andrii <lysenkoa.work@gmail.com>
Co-authored-by: Travis Walter <travis@automattic.com>

Initsogar committed:
Paid stats: update upsell copy in Stats modules (#97366)

* Update upsell copy in Stats modules

* Update upsell copy in Stats module including a link to support documentation.

* Update upsell copy for file downloads

* Update StatsCardUpsell to specify the upsell plan for WPCOM

* Update buttonLabel typing - adding TranslateResult type

lsl committed:
Enable paid stats v3 upsell in production config (#97310)

lsl committed:
Update copy on old modal for personal/premium upsell features (#97312)

chihsuan committed:
Fix WCCOM account creation flow UI (#97311)

* Fix flash placeholder in lost password screen

* Fix flash blue background

* Refactor UserStep component to use isWoo prop for WooCommerce-specific logic and skip account completion logic for Woo

* Add comment

* Update client/layout/masterbar/woo.scss

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

---------

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

ilyasfoo committed:
Update masterbar and jetpack header logo (#97318)

* Update masterbar and jetpack header logo

* Fix transform for old logo

ilyasfoo committed:
Update Woo branding for reset password pages (#97152)

* Add changes for reset password and fixes so rebrand styling update do not affect core profiler flow

* Use oauth class instead for more precise selector

* Update test

fushar committed:
Hosting dashboard: prevent showing invalid intermediate content when switching from Atomic to Simple (#97268)

arthur791004 committed:
Design Picker: Dedupe the themes (#97315)

* Design Picker: Dedupe the themes

* Address feedback

* Fix the order

arthur791004 committed:
Design Picker: Display the recommended themes (#97320)

* Design Picker: Add the Recommended themes section

* Design Picker: Don't show recommended themes if the selected categories have been changed

* Fix types

* Show no results only when the recommended designs section is hidden

* Fix track property

Initsogar committed:
Paid stats: update UTM and Devices on empty state to point to the right support URL (#97376)

* Update UTM module to conditionally render the support link

* Update Devices module to conditionally render the support link

* Update condition to use isJetpackSite selector

yashwin committed:
A4A > Feedback: Enable in-product feedback mechanism on staging (#97388)

bogiii committed:
Domain management: Match component design in the case where user has Mailboxes (#97252)

* Extend EmailComparison component to support adding container className

* Customize EmailComparison component to match figma design

* Customize EmailComparison component to match figma design

* Replace feature flag with usage context

* Rename constant to be a more accurate name

* Extend EmailPlanMailboxesList props to customize component

* Extend EmailPlan props to customize component

* Customize EmailPlan component based on feature flag

* Add context-all-domain-management style

* Show optional icon next to the mailbox title based on type

* Replace feature flag with context property

* Add mailbox icon based on context

* Rename constant to be a more accurate name

* Simplify header icon rendering

* Use destructor inside a function param

* Hide MailPoet upsell for all domains management context

* Add missing align-items property style

* Update children type definition

* Adjust section header text and button font size

* Update color variable

* Rearange css selector to avoid important operator

* Remove custom logic for getting email icon

* Fix alignment issue

madhusudhand committed:
Design Picker: Add free tier filter to Type category in design picker (#97323)

* add free filter to type filters

* remove free only filter component

* Just move free-only toggle into types and keep functionalites the same

* Update track events

---------

Co-authored-by: arthur <arthur.chu@automattic.com>

yashwin committed:
A4A > Feedback: Minor typo fix (#97397)

alshakero committed:
Login: Fix Safari autozoom (#97374)

alshakero committed:
Fix coupons in Stepper (#97359)

heavyweight committed:
Revert "Revert "Help Center: Get unread count when new message arrives (#97120)"" (#97402)

aidvu committed:
E2E Tests: Disable QUIC as it's sometimes unstable (#97403)

and browsers don't fallback :(

timur987 committed:
Return showing duration days back

Related to the commit 3fcd217255d07b960e163fccd5f0ce893aab7cd4.

AllTerrainDeveloper committed:
[Odie] We've missed to add the styling for error messages in the new help center experience (#97405)

mmtr committed:
Duplicate views: Redirect from Calypso to WP Admin (Posts, Testimonials, Portfolio) (#97231)

aidvu committed:
E2E: Fix the LOHP theme test (#97346)

Some back-and-forth to try and unflaky this test... Mix of race condition abort/goto and not being able to hover->click.

---------

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

alshakero committed:
Accessiblity: Fix Help Center portal injection (#97370)

Co-authored-by: Kosta <heavyweight@users.noreply.github.com>

alshakero committed:
Fix Stepper performance bug (#97354)

alshakero committed:
Stepper: Fix 404 error (#97407)

donnapep committed:
Conditionally update copy on migrate vs. import text page (#97125)

CGastrell committed:
Change Calypso Logo generator prompts (#97171)

* accept both remote and base64 urls and produce a well formed request for each

* move first logo generation prompt to backend

* use new flux request if a style is used as URL hash

* add image types and constants, add featuresControl to state

* remove style setting from URL hash

* modify hook functions to accept style for the request

* implement styles dropdown on logo generator prompt

* wee css fixes: spacing actions and make button logo (SVG) same color as button

alshakero committed:
Keep accent color untouched in Stepper (#97371)

sbarbosa commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -754,18 +780,44 @@ export default function CampaignItemDetails( props: Props ) {
 										<div>
 											<span className="campaign-item-details__label">
-												{ translate( 'Clicks' ) }
+												{ translate( 'Visitors' ) }
```
If I am not wrong, we decided to keep using clicks because they are a different metric than Visitors (not all clicks will end up being visitors). We didn't want to confuse users when they see visitors on the Stats page.

It may have changed since my AFK, but in the widget, we still use the clicks:
![Screenshot 2024-12-12 at 11 18 21 AM](https://github.com/user-attachments/assets/3d4a578e-e58b-42d2-a8af-25eca52a5efd)

The designs are not fully updated, but you can talk with Bart about this. I don't have the design P2 at hand, but I remember there was a discussion about this.

Commit:
Reader: Added tracks event for time spent reading the post. (#97389)

* Reader: Added tracks event for time spent reading the post.

* Track reading time when the current post changes.

sbarbosa reviewed (commented):
The code looks good @timur987.

I found a detail related to Clicks vs Visitors that I think we need to change. Also, probably not related to this PR, but I am getting a JS error in one of my campaigns:

![Screenshot 2024-12-12 at 11 35 45 AM](https://github.com/user-attachments/assets/7ab1d5ec-bb96-4814-8ff7-07c2ab3af93c)

Let me know if I need to open a different ticket for this specific error. It may be related to single-day campaigns.

escapemanuele committed:
Help Center: fire event when requesting support (#97419)

* Use selectedSite from storage

* Fire event when clicking on get support

* Remove unrelated code

andrii-lysenko committed:
a4a: reorder referrals table and change nume (#97377)

eoigal committed:
Reader: fix style in tags page sidebar (#97415)

* Reader: fix style in tags page sidebar

* fix horizontal line too

jwebbdev committed:
Paid Stats: Add expandable feature list to upsell (#97367)

* Add expandable feature list to upsell

* Add new images

* Hide overflow in right panel

* Add expandable insights upsell

* Update style on expand div

* Add gradient on upsell right panel

* Fix upsell border-radius

* Fix upsell image in mobile

* Update checkout feature list for stats

* Fix for stats rows breaking out

---------

Co-authored-by: Rafael Agostini <rafael.agostini@automattic.com>

alshakero committed:
Stepper: Move conservative lazy loading (#97326)

gabrielcaires committed:
Enable wrap words when needed (#97339)

alshakero committed:
Stepper: Fix domains step persistence (#97375)

oskosk committed:
My Home: Fix alignment on inspiration card call action button (#97422)



* Scope the styles for the reader-first-posts__nudge card

heavyweight committed:
Revert "Accessiblity: Fix Help Center portal injection" (#97425)

gabrielcaires committed:
Remove english gateway (#96502)

myhro committed:
Stats: improve tooltips positive/negative trends (#97364)

* Stats: refactor 'getNote()' function

Provide the note together with the overview data. This way, it will be
available regardless of whether the site has paid subscribers, as they
are handled differently.

* Stats: improve tooltips positive/negative trends

* Compared translated strings between notes/labels

michaeldcain committed:
Scroll to the top when Checkout loads (#97427)

travisw committed:
A4A: Enable in-product feedback on staging and production (#97372)

* enable in-product feedback on staging and production

* testing change for flaky test

* redo testing change for flaky test

* change back to intended code. the tests still failed with the noop code conifrming it is unrelated

rafaelgallani committed:
Revert "Fix Stepper performance bug (#97354)" (#97429)

This reverts commit 8c0021bd1986e49e9b9931fd23a50e3ad9cc0b1f.

sirbrillig committed:
Purchases: Convert me/purchases/utils to TS (#97297)

* Convert me/purchases/utils to TS

* Prevent calling purchase functions without purchase

* Export utils functions directly

* Add explicit return types to all util functions

sirbrillig committed:
composite-checkout: Add paymentMethod ID to errors logged by submit_button_load (#97432)

jwebbdev committed:
All date controls up to 30 days for basic plans (#97431)

vykes-mac committed:
Goals Screen: Save onboarding goals when creating site (#97426)

* include onboarding site goals in the site creation step

* gate goal saving behind the experiment feature flag

* prevent site goals from being overwritten with empty goals

a8ck3n committed:
Stats: Put 30-days summary behind paywall (#97387)

* Put 30 days summary behind paywall

* Update client/my-sites/stats/hooks/use-should-gate-stats.ts

* Update client/my-sites/stats/hooks/use-should-gate-stats.ts

Co-authored-by: Dognose <dognose24@users.noreply.github.com>

* Add import

---------

Co-authored-by: Louis Laugesen <louis.laugesen@gmail.com>
Co-authored-by: Dognose <dognose24@users.noreply.github.com>

p-jackson committed:
Goals First experiment should only be assigned in onboarding flow (#97438)

yashwin committed:
A4A > Agency Tier: Navigate user to the correct section of the KB article (#97385)

arthur791004 committed:
Design Picker: Update filter styles (#97393)

* Design Picker: Update filter styles

* Fix styles on single selection

* Remove border from the topic list

* Fix the behavior of the responsive toolbar

arthur791004 committed:
Design Picker: Update the track event properties (#97379)

* Design Picker: Update the track event properties

* Update tests

* Fix types

dognose24 committed:
Stats: Fix hourly views when the date control is gated (#97437)

* Stop forcing date range for period hour

* Stop bar clicking when date control is locked

epeicher committed:
Performance Profiler: Pass translate function to metrics when rendering (#97420)

rcrdortiz committed:
Sidebar Upsell Nudge: Improve UI  (#97278)

* Updated the UI according to the design agreed on

* Non-dismissible notices do not have a UI change

* UI changes now only apply to is-dismissible + is-compact

* Minor change in CSS selector

gmjuhasz committed:
Social Connections: Fix bug with service examples (#97332)

* Add all service ID to SERVICES_WITH_EXAMPLES

* Add remaining services

* Update client/sites/marketing/connections/service-examples.jsx

Co-authored-by: Manzoor Wani <manzoorwani.jk@gmail.com>

---------

Co-authored-by: Manzoor Wani <manzoorwani.jk@gmail.com>

arthur791004 committed:
Design Picker: Dedupe themes shown on the Recommended themes section (#97441)

yashwin committed:
A4A > Referrals: Implement commissions payment delayed banner (#97443)

andres-blanco committed:
Remove GA nudge in stats page (#97447)

Co-authored-by: Andrés Blanco <email@gmail.com>

timur987 committed:
Revert using "Clicks" instead of "Visitors"

Since we still use "Clicks" in DSP Widget, changing them to "Visitors" is not as consistent. So we will address that design change later.

timur987 committed:
Address the case of having less that 1 day

Otherwise, for campaigns that were started "today" we will see "0" in the UI which is not appropriate.

timur987 committed:
Overcome uPlot issue with infinite loop

Issue is "Range Error: Invalid array length",
https://github.com/leeoniya/uPlot/issues/827.

timur987 committed:
Merge remote-tracking branch 'origin/trunk' into add/revert-duration-days

timur987 commented:
@sbarbosa, thanks for your input. Indeed there was a "Range Error: Invalid array length" exception due to uPlot issue (https://github.com/leeoniya/uPlot/issues/827).

Due to that, I restricted data to be processed with uPlot to at least 2 elements (so minimum and maximum values are likely be different and due to that the infinite loop won't happen.

Also, I replaced "Visitors" with "Clicks" back.

Please take a look.

Here's a demo for infinite loop and how it happens:

https://github.com/user-attachments/assets/31b8b30f-3e5f-4b4e-864b-eb32af239698

sbarbosa reviewed (approved):
LGTM 👍

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17065342).
